MathUtils/src/lib/regression/LinearRegression.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2012-2021 Euclid Science Ground Segment
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 3.0 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "MathUtils/regression/LinearRegression.h"
+#include <ElementsKernel/Exception.h>
+
+namespace Euclid {
+namespace MathUtils {
+
+std::pair<double, double> linearRegression(const std::vector<double>& x, const std::vector<double>& y) {
+  if (x.size() != y.size()) {
+    throw Elements::Exception("linearRegression: x and y vectors must have the same size");
+  }
+  if (x.empty()) {
+    throw Elements::Exception("linearRegression: Can not do a regression over empty vectors");
+  }
+
+  double mean_x = 0;
```

Comment:
I would rather use double literal like 0.0 and not 0. This is probably meaningless for the null argument, but a cast from 1 to double might slightly different from 1.0.